src/util/probes.d

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2022 The Bitcoin Core developers
+   Distributed under the MIT software license, see the accompanying
+   file COPYING or http://www.opensource.org/licenses/mit-license.php. */
+
+/*
+
+   This file is part of the more involved process of using USDT probes on macOS
+   as described in the BUILDING CODE CONTAINING USDT PROBES section in the
+   dtrace(1) manpage https://www.unix.com/man-page/mojave/1/dtrace/
+
+   This defines the providers and specifies their probes in order to generate
+   the header file `probes.h` with the necessary tracepoints macros.
+
+   Types not supported by the D language cannot be used as part of the probe's
+   signature. On those occasions, temporary D-supported types are used to then
+   replaced in the generated header file with those that we actually need.
+
+   The unsupported types are specified with inline comments next to their
+   temporary D language supported replacement.
+
+*/
+
+provider net {
+        probe inbound_message(int64_t, char *, char *, char *, int64_t, unsigned char *);
+        probe outbound_message(int64_t, char *, char *, char *, int64_t, /* std::byte * */unsigned char *);
+};
```

Comment:
does `inbound_message` also need the `/* std::byte * */` comment? They both pass the same data.

Reply:
I actually made a mistake with the comment there, it's the other way around. `inbound_message` is the one that needs the change to `std::byte *`. `outbound_message` works as it is with `unsigned char *`.

Trying to build without changing the type at `inbound_message` results to

```
error: no matching function
...
/util/probes.h:46:13: note: candidate function not viable: no known conversion from 'CDataStream::value_type *' (aka 'std::byte *') to 'const unsigned char *' for 6th argument
```
That's why I changed that to `std::byte *`. Reading the docs, my understanding was also that the both pass the same data but it seems that the 6th arg of `inbound_message` uses the `CDataStream` class while `outbound_message` uses a `std::vector<unsigned char>`.

Reply:
Oh, you are right. Then we should reinterpret_cast `.data()` to `const unsigned char *` if it makes thinks easier for you (and is currently not conforming to the documentation).

Reply:
Great idea! `(unsigned char*)(msg.m_recv.data())` seems to work. But I am not sure if this has performance overhead.